wall_e/src/WalleModels/models.py

```diff
@@ -1,11 +1,74 @@
+# for type references to own class
+# https://stackoverflow.com/a/33533514
+from __future__ import annotations
+from typing import List
 import datetime
 import time
-
 import pytz
 from asgiref.sync import sync_to_async
 from django.conf import settings
 from django.db import models
 from django.forms import model_to_dict
+from .customFields import GeneratedIdentityField
+
+
+import logging
+logger = logging.getLogger('wall_e')
+
+
+class BanRecords(models.Model):
+    ban_id = GeneratedIdentityField(primary_key=True, always=True)
+    username = models.CharField(max_length=32, null=False)
+    user_id = models.BigIntegerField(null=False)
+    mod = models.CharField(max_length=32, null=True)
+    mod_id = models.BigIntegerField(null=True)
+    ban_date = models.BigIntegerField(null=True)
+    reason = models.TextField(null=False)
+    unban_date = models.BigIntegerField(null=True, default=None)
+
+    class Meta:
+        db_table = 'WalleModels_ban_records'
+
+    @classmethod
+    @sync_to_async
+    def insert_records(cls, records: List[BanRecords]) -> None:
+        """Adds entry to BanRecords table"""
+        for record in records:
+            record.save()
+
+    @classmethod
+    @sync_to_async
+    def insert_record(cls, record: BanRecords) -> None:
+        """Adds entry to BanRecords table"""
+        record.save()
+
+    @classmethod
+    @sync_to_async
+    def get_all_active_ban_user_ids(cls) -> List[int]:
+        """Returns list of user_ids for all currently banned users"""
+
+        return list(BanRecords.objects.values_list('user_id', flat=True).filter(unban_date=None))
+
+    @classmethod
+    @sync_to_async
+    def get_all_active_bans(cls) -> List[BanRecords]:
+        """Returns list of usernames and user_ids for all currently banned users"""
+
+        return list(BanRecords.objects.values('username', 'user_id').filter(unban_date=None))
+
+    @classmethod
+    @sync_to_async
+    def unban_by_id(cls, user_id: int) -> str:
+        """Set active=False for user with the given user_id. This representes unbanning a user."""
+        user = BanRecords.objects.get(user_id=user_id, unban_date=None)
```

Comment:
I'd feel better if you caught the exception that happens here in the unlikely event that maybe there is no `BanRecord` with those conditions.

I know it'd probably be unlikely, but its not impossible